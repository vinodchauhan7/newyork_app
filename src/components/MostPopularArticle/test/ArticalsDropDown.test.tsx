
import ArticalsSizeDropDown  from "../ArticleDropDown"
import { render, screen, fireEvent} from "@testing-library/react";
import '@testing-library/jest-dom';

const article = [
    {
      uri: "nyt://article/43f70c4a-a3d5-5ed5-9c67-9b0f34016398",
      url: "https://www.nytimes.com/2024/07/13/us/politics/doug-mills-trump-photo.html",
      id: 100000009570347,
      asset_id: 100000009570347,
      source: "New York Times",
      published_date: "2024-07-13",
      updated: "2024-07-14 13:05:44",
      section: "U.S.",
      subsection: "Politics",
      nytdsection: "u.s.",
      adx_keywords:
        "Assassinations and Attempted Assassinations;Presidential Election of 2024;United States Politics and Government;Photography;News and News Media;Trump, Donald J;Crooks, Thomas Matthew (2003-24);Mills, Doug;New York Times;Butler (Pa)",
      column: null,
      byline: "By Doug Mills",
      type: "Article",
      title:
        "A Times Photographer Who Was Feet Away From Trump Describes the Shooting",
      abstract:
        "“I hope I get the right shot. I hope I’m not shot myself,” said Doug Mills, who has been photographing presidents since 1983.",
      des_facet: [
        "Assassinations and Attempted Assassinations",
        "Presidential Election of 2024",
        "United States Politics and Government",
        "Photography",
        "News and News Media",
      ],
      org_facet: ["New York Times"],
      per_facet: [
        "Trump, Donald J",
        "Crooks, Thomas Matthew (2003-24)",
        "Mills, Doug",
      ],
      geo_facet: ["Butler (Pa)"],
      media: [
        {
          type: "image",
          subtype: "photo",
          caption:
            "Secret Service agents surround Mr. Trump immediately after reports of gunfire at his rally.",
          copyright: "Doug Mills/The New York Times",
          approved_for_syndication: 1,
          "media-metadata": [
            {
              url: "https://static01.nyt.com/images/2024/07/14/multimedia/13xp-mills/13xp-mills-thumbStandard.jpg",
              format: "Standard Thumbnail",
              height: 75,
              width: 75,
            },
            {
              url: "https://static01.nyt.com/images/2024/07/14/multimedia/13xp-mills/13xp-mills-mediumThreeByTwo210.jpg",
              format: "mediumThreeByTwo210",
              height: 140,
              width: 210,
            },
            {
              url: "https://static01.nyt.com/images/2024/07/14/multimedia/13xp-mills/13xp-mills-mediumThreeByTwo440.jpg",
              format: "mediumThreeByTwo440",
              height: 293,
              width: 440,
            },
          ],
        },
      ],
    }
];

test("Render Articals Dropdown Component", () => {
  const {container} = render(
    <ArticalsSizeDropDown 
        articalsSize={1}
        onClickSetArticalsSize={() => {}}
        popularArticals={article}
    />);
    fireEvent.change(screen.getByTestId("days"), { target: { value: 7 } })
    let options = screen.getAllByTestId('select-option7')
    expect(options[0].innerHTML).toEqual('Last 7 days');
    expect(container).toMatchSnapshot();
})