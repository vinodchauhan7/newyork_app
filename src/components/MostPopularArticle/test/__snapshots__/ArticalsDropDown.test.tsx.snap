// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Articals Dropdown Component 1`] = `
<div>
  <div
    class="w-full flex justify-between items-center mx-auto max-w-7xl px-6 lg:px-8"
  >
    <div>
      1
       Popular Articles from last 
      1
       
      day
    </div>
    <div
      class="flex items-center md:w-[30%]"
    >
      <label
        aria-label="days"
        class=" mb-2 text-sm font-medium text-gray-900 dark:text-white"
      >
        Select Days
      </label>
      <select
        class="bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block ml-5 w-[80%] p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
        data-testid="days"
        name="day"
      >
        <option
          value="1"
        >
          1
           selected
        </option>
        <option
          data-testid="select-option1"
          value="1"
        >
          Last 1 day
        </option>
        <option
          data-testid="select-option7"
          value="7"
        >
          Last 7 days
        </option>
        <option
          data-testid="select-option30"
          value="30"
        >
          Last 30 days
        </option>
      </select>
    </div>
  </div>
</div>
`;
